pkg/signing/signing.go

```diff
@@ -89,15 +87,64 @@ func sifAddSignature(fimg *sif.FileImage, groupid, link uint32, fingerprint [20]
 	return nil
 }
 
+// Copy-paste from sylabs/sif
+// datatypeStr returns a string representation of a datatype.
+func datatypeStr(dtype sif.Datatype) string {
+	switch dtype {
+	case sif.DataDeffile:
+		return "Def.FILE"
+	case sif.DataEnvVar:
+		return "Env.Vars"
+	case sif.DataLabels:
+		return "JSON.Labels"
+	case sif.DataPartition:
+		return "FS"
+	case sif.DataSignature:
+		return "Signature"
+	case sif.DataGenericJSON:
+		return "JSON.Generic"
+	case sif.DataGeneric:
+		return "Generic/Raw"
+	}
+	return "Unknown data-type"
+}
+
+func getDataPartitionToSign(fimg *sif.FileImage, dataType sif.Datatype) ([]*sif.Descriptor, error) {
+	sylog.Debugf("Looking for: %s partition to sign...", datatypeStr(dataType))
+	data, _, err := fimg.GetLinkedDescrsByType(uint32(0), dataType)
+	if err != nil && err != sif.ErrNotFound {
+		return nil, fmt.Errorf("failed to get descr for deffile: %s", err)
+	}
+	sylog.Debugf("Found %d partitions", len(data))
+
+	return data, nil
+}
+
 // descrToSign determines via argument or interactively which descriptor to sign
-func descrToSign(fimg *sif.FileImage, id uint32, isGroup bool) (descr []*sif.Descriptor, err error) {
-	descr = make([]*sif.Descriptor, 1)
+func descrToSign(fimg *sif.FileImage, id uint32, isGroup bool) ([]*sif.Descriptor, error) {
+	descr := make([]*sif.Descriptor, 1)
+	var err error
 
 	if id == 0 {
 		descr[0], _, err = fimg.GetPartPrimSys()
 		if err != nil {
 			return nil, fmt.Errorf("no primary partition found")
 		}
+
+		// signablePartitions is a list of all the signable datatypes.
+		signablePartitions := []sif.Datatype{
```

Comment:
nit picking: it might be more clear to call this "signableDatatypes" and call the loop variable "datatype" instead of s?